src/promptflow/promptflow/executor/_assistant_tool_invoker.py

```diff
@@ -0,0 +1,96 @@
+import os
+from dataclasses import dataclass
+from functools import partial
+from pathlib import Path
+from typing import Callable, Dict
+
+from promptflow.contracts.flow import InputAssignment, Node, ToolSource
+from promptflow.contracts.tool import ToolType
+from promptflow.exceptions import ErrorTarget
+from promptflow.executor._docstring_parser import DocstringParser
+from promptflow.executor._errors import UnsupportedAssistantToolType
+from promptflow.executor._tool_resolver import ToolResolver
+
+
+@dataclass
+class AssistantTool:
+    name: str
+    definition: dict
+    func: Callable
+
+
+class AssistantToolInvoker:
+    def __init__(self):
+        self._assistant_tools: Dict[str, AssistantTool] = {}
+
+    def load_tools(self, tools: list):
+        for tool in tools:
+            if tool["type"] in ("code_interpreter", "retrieval"):
+                self._assistant_tools[tool["type"]] = AssistantTool(name=tool["type"], definition=tool, func=None)
+            elif tool["type"] == "function":
+                function_tool = self._load_tool_as_function(tool)
+                self._assistant_tools[function_tool.name] = function_tool
+            else:
+                raise UnsupportedAssistantToolType(
+                    message_format="Unsupported assistant tool type: {tool_type}",
+                    tool_type=tool["type"],
+                    target=ErrorTarget.EXECUTOR,
+                )
+
+    def _load_tool_as_function(self, tool: dict):
+        tool_resolver = ToolResolver(working_dir=Path(os.getcwd()), need_connections=False)
```

Comment:
It might be more consistent to set working_dir as an init parameter?

Reply:
Refined.